### DISCUSSION TITLE
On Coding Bets

### DISCUSSION THREAD

**Discussion by robmoffat** [risk-first/website#72] [Team Posts]
Following on from the idea that [A Development Team's Job Is Balancing Risk](https://riskfirst.org/Purpose-Development-Team), I've written an article addressing the activity of [Coding](https://riskfirst.org/Coding-Bets), and pointing out that whenever we write code, we are addressing some risks in our project.

Specifically, I'm looking at:

- Writing New Features
- Refactoring
- Spike Solutions
- Coding a Minimum Viable Product

... there are probably some other examples it would be nice to cover - what do you think?

Please check it out [here](https://riskfirst.org/Coding-Bets) and let me know

cheers!

**Comment by hovancik** [comment]
I feel like there is an introduction missing...

**Comment by robmoffat** [comment]
You're right of course - I've added one now.  

I just hate it when the word count gets away from me.

**Comment by nfultz** [comment]
It's an interesting idea - seems like you are focusing on uncertainty *within* a task - usually I've seen this argued / presented as comparing between tasks (eg refactor vs features vs bugs) - and then there's usually a time component to think about as well (eg features now means more bugs later)

**Comment by robmoffat** [comment]
Hi Neal,

Yes, that’s right, it’s tasks individually here, although you’re spot on.  Just like placing a bet where you have an opportunity cost that your money can’t be placed on _other_ bets, if you pick one feature to deliver then there is the opportunity cost of all the work you can’t do.

I think I talk about your specific refactoring/building features example [here](https://riskfirst.org/Purpose-Development-Team), which looks a bit more closely at that whole this-or-that dynamic. 

I’ve also got in mind a third level of looking at this which is the _in task_ view, which is just notes at the moment.  That is, how does risk affect the way in which you go about a task?  

I’ve got some interesting examples around writing tests and debugging.  Could do with a few more ideas yet though.

**Comment by BaseMax** [comment]
Hey, https://riskfirst.org/Coding-Bets :wave:
